src/libxively/event_dispatcher/xi_time_event.c

```diff
@@ -0,0 +1,348 @@
+/* Copyright (c) 2003-2015, LogMeIn, Inc. All rights reserved.
+ * This is part of Xively C library. */
+
+#include "xi_time_event.h"
+
+/**
+ * @brief This part of the file implements time event functionality. This
+ * implementation assumes that the element type is always the xi_time_event_t.
+ *
+ * It uses the vector as a container type. The vector stores pointers to the time events.
+ * Time events are always sorted in the vector by the time event
+ *
+ */
+
```

Comment:
This is for ```static``` functions local to their compilation unit. This is not exposed API function that's why it has it's description in .c.